ethereum/executionlayer/src/test/java/tech/pegasys/teku/ethereum/executionlayer/MilestoneBasedExecutionJsonRpcMethodsResolverTest.java

```diff
@@ -55,11 +55,13 @@ void shouldProvideExpectedMethodsForBellatrix(
       EngineApiMethods method, Class<EngineJsonRpcMethod<?>> expectedMethodClass) {
     final Spec bellatrixSpec = TestSpecFactory.createMinimalBellatrix();
 
-    final MilestoneBasedExecutionJsonRpcMethodsProvider methodsProvider =
-        new MilestoneBasedExecutionJsonRpcMethodsProvider(bellatrixSpec, executionEngineClient);
+    final LocallySupportedEngineApiCapabilitiesProvider capabilitiesProvider =
+        new LocallySupportedEngineApiCapabilitiesProvider(bellatrixSpec, executionEngineClient);
+    final MilestoneBasedExecutionJsonRpcMethodsResolver methodsResolver =
+        new MilestoneBasedExecutionJsonRpcMethodsResolver(bellatrixSpec, capabilitiesProvider);
 
     final EngineJsonRpcMethod<Object> providedMethod =
-        methodsProvider.getMethod(method, Object.class);
+        methodsResolver.getMethod(method, Object.class);
```

Comment:
can something happen here where we dont have to receive object? not sure if its worth it... seems ugly though...